pkg/scheduler/scheduler_test.go

```diff
@@ -177,79 +183,113 @@ func TestScheduler(t *testing.T) {
 			expectErrorPod:   podWithID("foo", testNode.Name),
 			expectForgetPod:  podWithID("foo", testNode.Name),
 			eventReason:      "FailedScheduling",
-		}, {
+		},
+		{
+			name:        "deleting pod failed scheduling",
 			sendPod:     deletingPod("foo"),
 			algo:        mockScheduler{"", nil},
 			eventReason: "FailedScheduling",
 		},
 	}
 
-	for i, item := range table {
-		var gotError error
-		var gotPod *v1.Pod
-		var gotForgetPod *v1.Pod
-		var gotAssumedPod *v1.Pod
-		var gotBinding *v1.Binding
-		configurator := &FakeConfigurator{
-			Config: &Config{
-				SchedulerCache: &schedulertesting.FakeCache{
-					ForgetFunc: func(pod *v1.Pod) {
-						gotForgetPod = pod
-					},
-					AssumeFunc: func(pod *v1.Pod) {
-						gotAssumedPod = pod
-					},
-				},
-				NodeLister: schedulertesting.FakeNodeLister(
-					[]*v1.Node{&testNode},
-				),
-				Algorithm: item.algo,
-				GetBinder: func(pod *v1.Pod) Binder {
-					return fakeBinder{func(b *v1.Binding) error {
-						gotBinding = b
-						return item.injectBindError
-					}}
-				},
-				PodConditionUpdater: fakePodConditionUpdater{},
-				Error: func(p *v1.Pod, err error) {
-					gotPod = p
-					gotError = err
+	for _, item := range table {
+		t.Run(item.name, func(t *testing.T) {
+			testScheduler(
+				item.injectBindError,
+				item.sendPod,
+				item.algo,
+				item.expectErrorPod,
+				item.expectForgetPod,
+				item.expectAssumedPod,
+				item.expectError,
+				item.expectBind,
+				item.eventReason,
+				eventBroadcaster,
+				testNode,
+				t,
+			)
+		})
+	}
+}
+
+func testScheduler(
+	injectBindError error,
```

Comment:
As we discussed in https://github.com/kubernetes/kubernetes/pull/63665#issuecomment-393884049, let's keep this inline for now.